mne/simulation/tests/test_metrics.py

```diff
@@ -44,7 +44,8 @@ def test_metrics():
     stc_bad = stc2.copy().crop(0, 0.5)
     assert_raises(ValueError, source_estimate_quantification, stc1, stc_bad)
     stc_bad = stc2.copy()
-    stc_bad.times -= 0.1
+    stc_bad.tmin -= 0.1
+    stc_bad.update_times()
```

Comment:
The problem is that `.times` is also dependent on `._data`. Which means this still goes wrong:

    # Resample
    stc.tstep /= decim
    stc._data = stc._data[:, ::decim]

having to worry about the order of such things does for me negate the safety the properties are attempting to provide. Having to explicitly call `update_times()` makes this a much more obvious error:

    # Resample
    stc.tstep /= decim
    stc.update_times()
    stc._data = stc._data[:, ::decim]

IMO the real solution is just computing `.times` on the fly and take the few milliseconds performance hit. But if that is not desirable, than at making `update_times()` a public method seems like a good choice to me.